app/main.py

```diff
@@ -2,0 +1,20 @@
+import datetime
+from app.cafe import Cafe
+
+
+def go_to_cafe(friends: list, cafe: Cafe) -> str:
+    num_maskless = 0
+    for friend in friends:
+        if "vaccine" not in friend:
+            return "All friends should be vaccinated"
+        expiration_date = friend["vaccine"].get("expiration_date")
+        if expiration_date < datetime.date.today():
+            return "All friends should be vaccinated"
+        if not friend.get("wearing_a_mask", False):
+            num_maskless += 1
+
+    if num_maskless:
+        return (f"Friends should buy {num_maskless} "
+                f"mask{"s" if num_maskless > 1 else ""}")
```

Comment:
Code Efficiency: There's no need to check `if num_maskless:` before returning the message. If `num_maskless` is zero, the message will just say 'Friends should buy 0 masks', which is still accurate. Removing this check would make your code slightly more efficient.